src/main/java/ca/uhn/fhir/jpa/starter/service/HelperService.java

```diff
@@ -562,6 +590,53 @@ public ResponseEntity<?> getTableData(Long lastUpdated){
 		return new ResponseEntity<List<PatientIdentifierEntity>>(patientInfoResourceEntities,HttpStatus.OK);
 	}
 
+	public ResponseEntity<String> computeSyncTime(String practitionerRoleId,String env){
+		String organizationId = getOrganizationIdByPractitionerRoleId(practitionerRoleId);
+		List<Timestamp> facilityWiseTimestamps = new ArrayList<>();
+		notificationDataSource = NotificationDataSource.getInstance();
+
+		Pair<List<String>, LinkedHashMap<String, List<String>>> idsAndOrgIdToChildrenMapPair = fetchIdsAndOrgIdToChildrenMapPair(organizationId);
+
+		// Get the current date and time
+		Timestamp currentTimestamp = new Timestamp(System.currentTimeMillis());
+		long fiveDaysAgoMillis = currentTimestamp.getTime() - (5 * millisecondsInADay);
+		Timestamp fiveDaysAgoTimestamp = new Timestamp(fiveDaysAgoMillis);
+
+		List<LastSyncEntity> lastSyncData = notificationDataSource.fetchLastSyncEntitiesByOrgs(idsAndOrgIdToChildrenMapPair.first,env,ApiAsyncTaskEntity.Status.COMPLETED.name(),fiveDaysAgoTimestamp);
+
+		// Group the data by organization ID
+		Map<String, List<LastSyncEntity>> groupedData = lastSyncData.stream()
+			.collect(Collectors.groupingBy(LastSyncEntity::getOrgId));
+
+		// Sort each group by startDateTime
+		groupedData.values().forEach(list -> list.sort(Comparator.comparing(LastSyncEntity::getStartDateTime)));
+
+		// Loop through the grouped data
+		for (List<LastSyncEntity> entityList : groupedData.values()) {
+			if (!entityList.isEmpty()) {
+				// Get the last element in the list
+				LastSyncEntity lastEntity = entityList.get(entityList.size() - 1);
+
+				// Check if endDateTime is not null
+				if (lastEntity.getEndDateTime() != null) {
```

Comment:
even lastEntity can be null

Reply:
if lastEntity is null it will not be added to facilityWiseTimestamps